lib/create-http-client.js

```diff
@@ -17,38 +25,78 @@ import rateLimit from './rate-limit'
  * @prop {Object=} headers - Additional headers
  * @return {Object} Initialized axios instance
  */
-export default function createHttpClient (axios, httpClientParams) {
-  const {space, accessToken, insecure, host, defaultHostname, httpAgent, httpsAgent, proxy, retryOnError, timeout} = httpClientParams
-  let {headers} = httpClientParams
-  let [hostname, port] = (host && host.split(':')) || []
-  hostname = hostname || defaultHostname
-  port = port || (insecure ? 80 : 443)
-  let baseURL = `${insecure ? 'http' : 'https'}://${hostname}:${port}/spaces/`
-  if (space) {
-    baseURL += space + '/'
+export default function createHttpClient (axios, options) {
+  const defaultConfig = {
+    insecure: false,
+    retryOnError: true,
+    logEmitter: new EventEmitter(),
+    logHandler: (log) => console.log(log),
+    errorEmitter: new EventEmitter(),
+    errorHandler: (log) => console.error(log),
+    // Passed to axios
+    headers: {},
+    httpAgent: false,
+    httpsAgent: false,
+    timeout: 30000,
+    proxy: false
+  }
+  const config = Object.assign({}, defaultConfig, options)
+
+  if (!config.accessToken) {
+    throw new TypeError(`'accessToken' option is missing`)
   }
-  headers = headers || {}
-  headers['Authorization'] = 'Bearer ' + accessToken
+
+  // Set up log & error listeners
+  if (config.logHandler) {
+    config.logEmitter.addListener('log', config.logHandler)
+  }
+
+  if (config.errorHandler) {
+    config.errorEmitter.addListener('error', config.errorHandler)
+  }
+
+  // Construct axios baseURL option
+  const protocol = config.insecure ? 'http' : 'https'
+  const space = config.space ? `${config.space}/` : ''
+  let hostname = config.defaultHostname
```

Comment:
Since this differs from SDK to SDK, we can't set a default here. The implementation has to do it